### PR TITLE
Improve form configuration article. Fixes #690

### LINKED ISSUE #690: Dynamic return_to
This is less known feature of both html standard and our platform, yet very powerful that.

Not alot of people know that button can send value and decide on where user will land based on the button clicked. ;-)

```

    <button name="return_to" value="/onboarding/step2">Back</button>
    <button name="return_to" value="/onboarding/step4">Next</button>

```

### PR DESCRIPTION
**Pull request by pavelloz**

### CODE REVIEW

File: app/views/pages/developer-guide/forms/form-configuration.liquid [hunk 1]
```diff
@@ -167,16 +174,53 @@ Head to the [Default Payload Tutorial](/developer-guide/forms/default-payload) t
...
-String, message that is passed to the `context.flash.notice` object when the form is valid.
+String, message that is passed to the `context.flash.notice` object when the form is valid. This type of message is usually used to notify user about success of an operation and is used by the system only if form submitted succeeded.
 Learn more in a separate tutorial about [Flash Messages](/developer-guide/forms/flash-messages).
 
-### redirect_to
-String, defines path of the page rendered after a successful form submission. Example: `/welcome`
+### return_to
+String, defines path of the page rendered after a successful form submission. Example: `/welcome`. You can use liquid to determine what the value should be.
+This property also can be set up during the form submission by setting value of `return_to` parameter.
+For example you can have two submitting buttons in your form, with different `return_to` values to send user into different places depending on which one has been clicked.
```
Comment (pavelloz) [inline app/views/pages/developer-guide/forms/form-configuration.liquid]:
```suggestion
For example you can have two submitting buttons in your form, with different `redirect_to` values to send user into different places depending on which one has been clicked.
```

File: app/views/pages/developer-guide/forms/form-configuration.liquid [hunk 2]
```diff
@@ -167,16 +174,53 @@ Head to the [Default Payload Tutorial](/developer-guide/forms/default-payload) t
...
 Learn more in a separate tutorial about [Flash Messages](/developer-guide/forms/flash-messages).
 
 ### flash_notice
-String, message that is passed to the `context.flash.notice` object when the form is valid.
+String, message that is passed to the `context.flash.notice` object when the form is valid. This type of message is usually used to notify user about success of an operation and is used by the system only if form submitted succeeded.
 Learn more in a separate tutorial about [Flash Messages](/developer-guide/forms/flash-messages).
 
-### redirect_to
-String, defines path of the page rendered after a successful form submission. Example: `/welcome`
+### return_to
```
Comment (pavelloz) [inline app/views/pages/developer-guide/forms/form-configuration.liquid]:
```suggestion
### redirect_to
```

File: app/views/pages/developer-guide/forms/form-configuration.liquid [hunk 3]
```diff
@@ -167,16 +174,53 @@ Head to the [Default Payload Tutorial](/developer-guide/forms/default-payload) t
...
+String, defines path of the page rendered after a successful form submission. Example: `/welcome`. You can use liquid to determine what the value should be.
+This property also can be set up during the form submission by setting value of `return_to` parameter.
```
Comment (pavelloz) [inline app/views/pages/developer-guide/forms/form-configuration.liquid]:
```suggestion
This property also can be set up during the form submission by setting value of `redirect_to` parameter.
```

File: app/views/pages/developer-guide/forms/form-configuration.liquid [hunk 4]
```diff
@@ -167,16 +174,53 @@ Head to the [Default Payload Tutorial](/developer-guide/forms/default-payload) t
...
+For example you can have two submitting buttons in your form, with different `return_to` values to send user into different places depending on which one has been clicked.
+
+For example, onboarding step 3 could have this code if we wanted to save progress (by submitting form) that user made on the form and go back/forward afterwards:
+
+```html
+<button name="return_to" value="/onboarding/step2">Back</button>
```
Comment (pavelloz) [inline app/views/pages/developer-guide/forms/form-configuration.liquid]:
```suggestion
<button name="redirect_to" value="/onboarding/step2">Back</button>
```

File: app/views/pages/developer-guide/forms/form-configuration.liquid [hunk 5]
```diff
@@ -167,16 +174,53 @@ Head to the [Default Payload Tutorial](/developer-guide/forms/default-payload) t
...
+<button name="return_to" value="/onboarding/step4">Next</button>
```
Comment (pavelloz) [inline app/views/pages/developer-guide/forms/form-configuration.liquid]:
```suggestion
<button name="redirect_to" value="/onboarding/step4">Next</button>
```

**Review by diana-lakatos — APPROVED** [review]
Reviewed and edited, thank you!

**Pull request merged**